src/tslint.ts

```diff
@@ -42,17 +44,46 @@ class Linter {
 
     private fileName: string;
     private source: string;
+    private program: ts.Program;
     private options: ILinterOptions;
 
-    constructor(fileName: string, source: string, options: ILinterOptionsRaw) {
+    public static getProgram(configFile: string, projectDirectory?: string): ts.Program {
```

Comment:
maybe `createProgram`? it should be clear that this is an expensive operation, not just a getter of a field